src/strategies/flex_window.py

```diff
@@ -84,170 +86,321 @@ def step(self, event_list=[]):
                 if type(event) == events.GridOperatorSignal:
                     # update GC info
                     cur_max_power = event.max_power or cur_max_power
-                    cur_window = event.window or cur_window
+                    cur_window = event.window
                 elif type(event) == events.EnergyFeedIn:
                     cur_feed_in[event.name] = event.value
-                # vehicle events ignored (use vehicle info such as estimated_time_of_departure)
+                    # vehicle events ignored (use vehicle info such as estimated_time_of_departure)
 
             # get (predicted) external load
-            if timestep_idx == 0:
-                # use actual external load
-                ext_load = gc.get_current_load()
-                # add battery power (sign switch, as ext_load is subtracted)
-                ext_load -= avail_bat_power
+            # if timestep_idx == 0:
+            #     # use actual external load
+            #     ext_load = gc.get_current_load()
+            # else:
+                # get external load
+            if self.EXT_LOAD == "perfect_foresight":
+                ext_load = self.events.get_external_load_perfect_forseight(
+                    start_time=cur_time, interval=self.interval
+                ) - sum(cur_feed_in.values())
```

Comment:
still not certain what this is trying to do - are you just interested in the current/next GridOperator and FeedIn events? Why not access self.events directly? You could save the actual values when iterating over the events (around l. 86 and 90).

Reply:
because the extenal load avents are not incuded in future_events. I'm not sure why. But I agree, it should be easier if they were

Reply:
the external load events are not in future_events. I don't know why that is. But I agree, it should work easier, still trying to find out why they are not included in future events..